src/main/java/com/datastax/cdm/properties/KnownProperties.java

```diff
@@ -132,6 +132,15 @@ public enum PropertyType {
         defaults.put(MAX_RETRIES, "0");
     }
 
+    //==========================================================================
+    // Error handling
+    //==========================================================================
+    public static final String TOKEN_RANGE_EXCEPTION_DIR = "spark.tokenRange.exceptionDir"; // aaa/bbb
+    public static final String TOKEN_RANGE_EXCEPTION_FILE = "spark.input.partitionFile"; // aaa/bbb/filename
+    static {
+            types.put(TOKEN_RANGE_EXCEPTION_DIR, PropertyType.STRING);
```

Comment:
Can we combine these two properties into one to reduce properties? The code should be able to easily figure the `dir` part from the full file-path.

Reply:
TOKEN_RANGE_EXCEPTION_FILE is removed now. Refactored to use the exiting class DiffPartitionsFromFile.scala for reading partition ranges for diff exceptions as well.